src/data/moves/move-attrs/torment-attr.ts

```diff
@@ -0,0 +1,21 @@
+import { BattlerTagType } from "#enums/battler-tag-type";
+import type { EnemyPokemon } from "#field/enemy-pokemon";
+import type { Pokemon } from "#field/pokemon";
+import { AddBattlerTagAttr } from "#moves/add-battler-tag-attr";
+import type { Move } from "#moves/move";
+
+/**
+ * Attribute to apply the effects of {@link https://bulbapedia.bulbagarden.net/wiki/Torment_(move) | Torment}.
+ * Prevents the target from using the same move twice in a row.
+ * @extends AddBattlerTagAttr
+ */
+export class TormentAttr extends AddBattlerTagAttr {
+  constructor() {
+    super(BattlerTagType.TORMENT, false, { failOnOverlap: true });
```

Comment:
```suggestion
  constructor(isAttack = false) {
    super(BattlerTagType.TORMENT, false, { failOnOverlap: !isAttack });
```
For G_MAX_MELTDOWN, assuming this is the same as other similar moves where it shouldn't fail if it's an attacking move
Would it also need the `lastHitOnly` option because of that? In which case
```suggestion
  constructor(isAttack = false) {
    super(BattlerTagType.TORMENT, false, {
      failOnOverlap: !isAttack,
      lastHitOnly: isAttack,
    });
```